lib/components/SelectionControl/ToggleSwitch/index.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useEffect, useState } from "react";
+import "./toggle-switch.scss";
+import clsx from "clsx";
+
+export interface SwitchToggleProps {
+    disabled?: boolean;
+    defaultChecked?: boolean;
+    onChange?: (isEnabled: boolean) => void;
+    className?: string;
+    id?: string;
+    onFocus?: () => void;
+    onBlur?: () => void;
+    checked?: boolean;
+}
+
+const ToggleSwitch = ({
+    disabled = false,
+    onChange,
+    className,
+    defaultChecked,
+    onFocus,
+    onBlur,
+    id,
+    checked,
+}: SwitchToggleProps) => {
+    const [isEnabled, setIsEnabled] = useState(defaultChecked || false);
+
+    useEffect(() => {
+        if (checked !== undefined) {
+            setIsEnabled(checked);
+        }
+    }, [checked]);
+
+    const toggleSwitch = () => {
+        if (!disabled) {
+            if (checked === undefined) {
+                const toggleEnable = !isEnabled;
+                setIsEnabled(toggleEnable);
+                onChange?.(toggleEnable);
+            } else {
+                onChange?.(!checked);
+            }
+        }
+    };
+
+    const handleKeyDown = (event: React.KeyboardEvent) => {
+        if (event.key === "Enter" || event.key === " ") {
+            event.preventDefault();
+            toggleSwitch();
+        }
+    };
+
+    return (
+        <div
+            className={clsx("toggle-switch", className, { disabled })}
+            tabIndex={disabled ? -1 : 0}
+            onClick={toggleSwitch}
+            onKeyDown={handleKeyDown}
+            role="button"
+            aria-pressed={isEnabled}
+            onFocus={onFocus}
+            id={id}
+            onBlur={onBlur}
+            aria-disabled={disabled}
+        >
+            <div
+                className={clsx("toggle-switch__slider", {
+                    __enabled: isEnabled,
```

Comment:
I suggest putting all the styles of `__enabled` as default styles inside `toggle-switch__slider`, get rid of `__enabled`, and use `isDisabled` for local state instead of `isEnabled`?
This way we will only use `disabled` className (without unnecessary `__`) to override default enabled styles when the element is `disabled` or `isDisabled`? 
In any case, we don't need `__` prefix for these classNames :)

Reply:
I linked disabled with button disabled pseudo class now . It just enabled that check if toggle is on or off.